src/modules/pijul.rs

```diff
@@ -0,0 +1,251 @@
+use unicode_segmentation::UnicodeSegmentation;
+
+use super::{Context, Module, ModuleConfig};
+
+use crate::configs::pijul::PijulConfig;
+use crate::formatter::StringFormatter;
+
+/// Creates a module with the Pijul channel in the current directory
+///
+/// Will display the channel lame if the current directory is a pijul repo
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let is_repo = context
+        .try_begin_scan()?
+        .set_folders(&[".pijul"])
+        .is_match();
+
+    if !is_repo {
+        return None;
+    }
+
+    let mut module = context.new_module("pijul");
+    let config: PijulConfig = PijulConfig::try_load(module.config);
+
+    // We default to disabled=true, so we have to check after loading our config module.
+    if config.disabled {
+        return None;
+    };
+
```

Comment:
For truncation, please follow the approach in the meson module, using the functions from `crate::utils::truncate`.

Reply:
I thought there ought to have been a better way, but I didn't look much outside the hg_branch module to find it.